tests/functional/src/main/java/org/mule/functional/classloading/isolation/maven/dependencies/TransitiveDependenciesFilterBuilder.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.classloading.isolation.maven.dependencies;
+
+import org.mule.functional.classloading.isolation.maven.MavenArtifact;
+import org.mule.functional.util.TruePredicate;
+
+import java.util.function.Predicate;
+
+/**
+ * A definition for the filtering strategy to be used in {@link DependencyResolver}
+ *
+ * @since 4.0
+ */
+public final class TransitiveDependenciesFilterBuilder
+{
+    private Predicate<MavenArtifact> predicate = new TruePredicate<>();
+    private boolean traverseWhenNoMatch = false;
+
+    /**
+     * Creates a new instance, only visible to package scope so the {@link ConfigurationBuilder} is the only one
+     * that should call this constructor.
+     *
+     * @param predicate
+     */
+    TransitiveDependenciesFilterBuilder(Predicate<MavenArtifact> predicate)
+    {
+        this.predicate = predicate;
+    }
+
+    /**
+     * Public constructor, accessible by clients of this API.
+     */
+    public TransitiveDependenciesFilterBuilder()
+    {
+    }
+
+    /**
+     * {@link Predicate<MavenArtifact>} to be used to filter which transitive dependencies should be included.
+     *
+     * @param predicate
+     * @return this
+     */
+    public TransitiveDependenciesFilterBuilder match(Predicate<MavenArtifact> predicate)
+    {
+        this.predicate = predicate;
+        return this;
+
+    }
+
+    /**
+     * Defines that if a transitive dependency does not match the predicate, it should not be included but
```

Comment:
The name traverseWhenNoMatch does not represents what that flag does. To me, it is more like "evaluateTransitiveDependenciesWhenPredicateFails" or similar